hrm-domain/hrm-core/case/caseSection/caseSectionService.ts

```diff
@@ -243,12 +254,16 @@ export const getMultipleCaseTimelines = async (
     parseInt(offset),
   );
   const timelines: Record<string, TimelineActivity<any>[]> = {};
-  for (const { caseId, ...activity } of dbResult.activities) {
+  for (const { caseId, ...activityEntry } of dbResult.activities) {
     timelines[caseId] = timelines[caseId] || [];
-    timelines[caseId].push({
-      ...activity,
-      activity: sectionRecordToSection(activity.activity),
-    });
+    if (isContactRecordTimelineActivity(activityEntry)) {
+      timelines[caseId].push({
+        ...activityEntry,
+        activity: contactRecordToContact(activityEntry.activity),
+      });
+    } else {
+      timelines[caseId].push(activityEntry);
+    }
```

Comment:
I find easier to read this stuff when written as `Array.map`s or family. Not asking for this to change, but is there a reason why you prefer this mutations approach (I've seen it in the last few PRs)?